text/0002-searchable-display-attributes-behaviors.md

```diff
@@ -0,0 +1,255 @@
+ TBD: What should be the position of `titre`?
+	- same position as `title` (thus allowing attributes to exists at the same position like Algolia)
+	- added at the last position
+
+#### `searchableAttributes` & `displayedAttributes`
+
+Being searchable and displayed are 2 different concept, and there should not be interactions between the two: a field can be searchable and not displayed, or displayed and not searchable, etc.
+
+#### add a new configuration `attributesPosition`?
+
+Adding a new configuration `attributesPosition` that link all attributes present in a document  to a position makes it possible to keep track of the position without caring about whether a field is searchable or displayed. This also allows to keep track of all the fields that have been found in documents in the database.
+
+
+The default order is then **always** the order a field was in a document when it was first found.
+
+##### Content of `attributesPosition`
+
+The role of `attributesPosition` is to link each field to a position in a document. Hence, it contains all the attributes that have been found while indexing documents and ascociate them with a position. By default, this position is the position the field had when it was first encountered in a document.
+
+##### Addition of new fields
+
+When a new field is discovered, it is given the position it had in the document where it was found. It is ok to have two fields with the same position, and they are considered equal according to the `Attributes` criterion.
+
+e.g: In this example is makes sense that this is the default:
+
+```json
+/// documents
+[
+	{
+		"id": 1,
+		"title": "the title"
+	},
+	{
+
+		"id": 2,
+		"titre": "le titre"
+	}
+]
+```
+```json
+/// attributesPosition
+{
+	"id": 0,
+	"title": 1,
+	"titre": 1
+}
+```
+
+##### Updating the `attributesPosition`
+
+There are 2 kinds of operations we need to consider:
+
+1) **Deletion**: `attributesPosition` are essential to the functionning of meilisearch. Their deletion should not be allowed. However they can be cleaned:
+calling `DELETE /indexes/:index_uid/attributes_position` triggers a reindexation, and only the fields present in the documents are kept. e.g:
+if my `attributesPosition` look like this: 
+```json
+/// attributesPosition
+{
+	"id": 0,
+	"title": 1,
+	"titre": 1
+}
+```
+and thoses are all of my documents are:
+```json
+/// documents
+[
+	{
+		"id": 1,
+		"title": "the title"
+	},
+	{
+
+		"id": 2,
+		"title": "the title2"
+	}
+]
+```
+
+then a call to `DELETE /indexes/:index_uid/attributes_position` will result in a re-indexation and `attributesPosition` will look like this: 
+```json
+/// attributesPosition
+{
+	"id": 0,
+	"title": 1
+}
+```
+
+2) **Updates**: As we have seen before, we don't want to allow deletion of those fields unless we are sure that this field does not exist in any documents. That would imply that trying to update the `attributesPosition` with a missing value is not allowed.
```

Comment:
what is the maximum number? Can it have negative values? Can there be gaps between positions?

Reply:
this is TBD, it will probably be bound by the type of the value we use to store the information, this is an implementation detail I don't have the answer to now. Gaps are tolerated.